mrmustard/lab/abstract/transformation.py

```diff
@@ -181,59 +197,52 @@ def d_vector_dual(self) -> Optional[RealVector]:
         if (d := self.d_vector) is None:
             return None
         if (Xdual := self.X_matrix_dual) is None:
-            return d
-        return math.matmul(Xdual, d)
+            return -d
+        return -math.matmul(Xdual, d)  # TODO: need minus sign?
```

Comment:
D^\dagger(\alpha) = D(-\alpha), so I think there is a minus sign.

Reply:
Perfect, thanks for confirming!